frontend/packages/dev-console/src/components/AddPage.tsx

```diff
@@ -17,11 +25,7 @@ export interface AddPageProps {
 }
 
 interface ResourcesType {
-  deploymentConfigs?: K8sResourceKind;
-  deployments?: K8sResourceKind;
-  daemonSets?: K8sResourceKind;
-  statefulSets?: K8sResourceKind;
-  knativeService?: K8sResourceKind;
+  [x: string]: K8sResourceKind & { data?: K8sResourceKind[] };
 }
 interface EmptyStateLoaderProps {
   resources?: ResourcesType;
```

Comment:
how about using the below type and remove the `ResourceType` altogether?
```suggestion
  resources?: TopologyDataResources;
```